### ISSUE TITLE
📖 Defining Guidelines to write Issues on Github

### ISSUE THREAD

**Issue by AntoinePlu** [followanalytics/commons#1]
### Context
After what we started on Jira, and since it's more openned for us to write 💩 in issues, it could be great to define guidelines. 📖

### How to write Issues on GitHub

**Comment by AntoinePlu** [comment]
Do you have other things in your mind @followanalytics/all ?

**Comment by jjoaquim** [comment]
More than the `How to fix` I'd say `Expected result` or something like that

**Comment by ook** [comment]
Simply `How` is enough to my eyes.

**Comment by aw2611** [comment]
I'd say `Expected Result` (or `Expected Behavior`) too, as Issues can be created both for a bug fix and for a new feature.

**Comment by agardez** [comment]
For structure I think the options apply either to bug or new features, we should specify this so that people don't try to fit in a model that doesn't work.

For the first one, shouldn't the Why be before the What? And be named context? :)
Seems more logical to me to explain the thing that doesn't work or should work, and then explain precisely what needs to be done in the card (maybe we should also make it clearer: context / what to do)

Also, for the notes: do not create an issue for something that's simply an idea of improvement or a vague new feature.

**Comment by aw2611** [comment]
It depends for bugs, it makes sens that "What" is before "Why".
> Example: 
> **What ?**
> Filters count is not well calculated
> **Why ?**
> After investigation, it seems that it is because we check the equality of attributes on all sessions for a device and not only for the last session of a device.

**Comment by bankair** [comment]
Trying to fit the bug and improvements in the same template is probably not a good idea.

**Comment by agardez** [comment]
Yeah, makes it clearer.

**Comment by AntoinePlu** [comment]
Why not? I mean, it works for millions of people, why not us?

1. We should start by defining the context. It works in every cases and it's a good way to be more accurate.
2. Describe your problem, your improvement or the next feature
3. What behavior is expected or the next step to move on
4. Test

**Comment by agardez** [comment]
For a bug, people won't answer the 4 questions.
1 + 2 = how to reproduce (browser, while showing to a client with bad connection, etc.)
3 + 4 = Expected (to test the resolution of the bug, you check that the expected behavior is there).

For a new feature or improvement, Context (1) is important ( = the Why in Alex's message).
Then the description (2) is the What. This is pretty close to #3, unless 3 is about more detailed implementation.
4 will be in the What for a lot of cards. 

My point is: If the template is too large, it won't be followed, or not properly, and will be more of a constraint.
Personally I like cutting into 3 or 4 sections for a new feature, but it's not the case for everyone and every improvement issues.

So: we could add an optional How to test step for new features/improvements.

**Comment by aw2611** [comment]
Guys, please **use Markdown**. Call me a pain in the ***, but I can't read a 15-line comment without formatting.

**Comment by agardez** [comment]
*(I edited my previous comment, hope you like it <3 )*

**Comment by aw2611** [comment]
:heart: @agardez

**Comment by AntoinePlu** [comment]
Sorry but with what you said @agardez @bankair I don't see if you want to have two different templates or not.

Please make it clear

**Comment by agardez** [comment]
Yes, as said by @bankair a template for bug, another for improvements / new features.

I'm just saying that we could add **how to test** to the new features template as an optional step, when it makes sense.
